### PR TITLE
Add PaLM vectorization source

### PR DESCRIPTION
**Pull request by theunstableman**

### CODE REVIEW

File: src/vectors.js [hunk 1]
```diff
@@ -15,6 +15,8 @@ async function getVector(source, text) {
             return require('./openai-vectors').getOpenAIVector(text);
         case 'transformers':
             return require('./embedding').getTransformersVector(text);
+        case 'palm':
+            return require('./embedding').getPaLMVector(text);
```
Comment (Cohee1207) [inline src/vectors.js:19]: The file reference is wrong

File: src/palm-vectors.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    const jsonifiedresp = JSON.parse(jsonString);
+  
+    // Access the "value" dictionary
+    const vector = jsonifiedresp.embedding.value;
+
+    return vector;
+}
+
+module.exports = {
+    getOpenAIVector,
```
Comment (Cohee1207) [inline src/palm-vectors.js:43]: The export function doesn't exist

File: src/palm-vectors.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    });
+
+    if (!response.ok) {
+        const text = await response.text();
+        console.log('PaLM request failed', response.statusText, text);
+        throw new Error('PaLM request failed');
+    }
+
+    const data = await response.json();
+    const jsonifiedresp = JSON.parse(jsonString);
```
Comment (Cohee1207) [inline src/palm-vectors.js:34]: You don't need this, just use `data` from the response

**Comment by theunstableman** [comment]
AW SHIT

**Comment by theunstableman** [comment]
ILL FIX IT

**Comment by theunstableman** [comment]
OK OK ILL RESUBMIT WITH FIXES

**Comment by 003nithin** [comment]
src/vectors.js
src/vectors.js
src/vectors.js

**Pull request closed without merging**